tests/src/Unit/ServerServiceTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+
+namespace Wsdl2PhpGenerator\Tests\Unit;
+
+
+use Wsdl2PhpGenerator\ClassGenerator;
+use Wsdl2PhpGenerator\Config;
+use Wsdl2PhpGenerator\Service;
+use Wsdl2PhpGenerator\ServerService;
+use Wsdl2PhpGenerator\Tests\Functional\FunctionalTestCase;
+
+/**
+ * Unit test for configuration of the SoapServer.
+ *
+ * This is functional test case as we have to create an instance of the service class. This forces us to have a valid
+ * WSDL which the server service class can be generated from.
+ */
+class ServerServiceTest extends CodeGenerationTestCase
+{
+
+    protected $namespace = 'SoapClientTest';
+
+    // Use our mock soap client. It allows to retrieve the configuration it was passed.
+    protected $soapclientClass = '\Wsdl2PhpGenerator\Tests\Mock\SoapClient';
+
+    // Example Wsdl path.
+    protected $wsdl = '/tmp/some.wsdl';
+
+    // Example options which can be passed as options to a \SoapClient instance.
+    protected $soapclientOptions = array(
+        'features' => SOAP_SINGLE_ELEMENT_ARRAYS,
+        'cache_wsdl' => WSDL_CACHE_NONE,
+    );
+
+    // Use our mock soap server. It allows to retrieve the configuration it was passed.
+    protected $soapserverClass = '\Wsdl2PhpGenerator\Tests\Mock\SoapServer';
+
+    // Example options which can be passed as options to a \SoapServer instance.
+    protected $soapserverOptions = array(
+        'features' => SOAP_SINGLE_ELEMENT_ARRAYS,
+        'cache_wsdl' => WSDL_CACHE_NONE,
+    );
+
+    /**
+     * Test handling of SoapServer configuration.
+     */
+    public function testSoapConfig()
+    {
+        $clientConfig = new Config(array(
+                'inputFile' => $this->wsdl,
+                'outputDir' => null,
+                'namespaceName' => $this->namespace,
+                'soapClientClass' => $this->soapclientClass,
+                'soapClientOptions' => $this->soapclientOptions,
+            ));
+
+        $clientService = new Service($clientConfig, 'TestClientService', array(), 'Service description');
+        $this->generateClass($clientService, $this->namespace);
+
+        $this->assertClassExists('TestClientService', $this->namespace);
```

Comment:
No, as the generated server class depends on the the classmap of the client class, we have to first create the client class before creating the server class. But if I duplicate the classmap as you  proposed, I can remove this part.